src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py

```diff
@@ -307,6 +307,10 @@ def validate_authentication_type(namespace, formula=None):
                 "incorrect usage for authentication-type 'password': "
                 "[--admin-username USERNAME] --admin-password PASSWORD")
 
+        # Respect user's provided saved secret name for password authentication
+        if namespace.saved_secret:
+            namespace.admin_password = "[[{}]]".format(namespace.saved_secret)
```

Comment:
Is this double bracket syntax a service-side construct?  

Either way, there should be validation that errors out if the user provides --admin-password AND --saved-secret.  The usage string error message should be updated as well. Same applies for SSH